action-server/covidflow/utils/persistence.py

```diff
@@ -1,89 +1,115 @@
+
+    return False
 
-    if is_test_phone_number(reminder.phone_number):
-        logger.info("555 number: not saving reminder to database")
-        return True
 
+def ci_enroll(slot_values: Dict[Text, Any]):
     session = session_factory()
     try:
-        session.add(reminder)
-        session.flush()
+        if _is_test_session(session, slot_values):
+            return
 
-        assessment = Assessment.create_from_slot_values(reminder.id, slot_values)
-        session.add(assessment)
+        reminder = _save_reminder(session, slot_values)
+        session.flush()
+        assessment = _save_assessment(session, slot_values, reminder.id)
 
         session.commit()
 
-        # the following is temporary and will allow us to get the hashids of the new reminders in the logs
         hashed_id = encode_reminder_id(reminder.id)
-        logger.info(f"Created new reminder (hashed_id={hashed_id}): {reminder}")
-
-        return True
+        logger.info("Saved reminder to database", hashed_id=hashed_id)
+        logger.debug("Saved assessment to database", assessment=assessment)
     except:
-        logger.exception("Could not save reminder instance")
         session.rollback()
-        return False
+        raise
     finally:
         session.close()
 
 
 def cancel_reminder(slot_values: Dict[Text, Any]):
-    reminder_id = get_reminder_id(slot_values)
-
     session = session_factory()
     try:
+        if _is_test_session(session, slot_values):
+            return
+
+        reminder_id = _get_reminder_id(slot_values)
         reminder = session.query(Reminder).get(reminder_id)
         reminder.is_canceled = True
+
         session.commit()
+
+        logger.debug("Canceled reminder", reminder=reminder)
     except:
-        logger.exception("Could not cancel reminder")
         session.rollback()
+        raise
     finally:
         session.close()
 
 
-def store_assessment(slot_values: Dict[Text, Any]):
-    metadata = slot_values.get(METADATA_SLOT, {})
-
-    if REMINDER_ID_METADATA_PROPERTY not in metadata:
-        raise KeyError(f"Missing {REMINDER_ID_METADATA_PROPERTY} in slot values")
-
-    hashed_id = metadata.get(REMINDER_ID_METADATA_PROPERTY, None)
-    reminder_id = decode_reminder_id(hashed_id)
+def save_assessment(slot_values: Dict[Text, Any], reminder_id=None):
+    logger.debug("Saving assessment")
 
     session = session_factory()
     try:
-        session.add(Assessment.create_from_slot_values(reminder_id, slot_values))
+        if _is_test_session(session, slot_values):
+            return
+
+        reminder_id = reminder_id or _get_reminder_id(slot_values)
+        assessment = Assessment.create_from_slot_values(reminder_id, slot_values)
+        session.add(assessment)
+
         session.commit()
+
+        logger.debug("Saved assessment", assessment=assessment)
     except:
-        logger.exception("Could not save assessment")
         session.rollback()
+        raise
```

Comment:
If you make it raise an error here, then you should catch it in daily_ci_assessment_common and do nothing with it. If we can't save one day's assessment, it should not disrupt the dialogue. Raising an error will stop the action server, the form will fail, and the dialogue will do weird things. We don't want that.

Reply:
Good point, I missed that one. Will also add try except for `cancel_reminder`